core/apple/test/NSOutputStreamSinkTest.kt

```diff
@@ -0,0 +1,45 @@
+package kotlinx.io
+
+import kotlinx.cinterop.IntVar
+import kotlinx.cinterop.pointed
+import kotlinx.cinterop.reinterpret
+import kotlinx.cinterop.value
+import platform.Foundation.*
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class NSOutputStreamSinkTest {
+    @Test
+    fun nsOutputStreamSink() {
```

Comment:
The test failed in CI w/ 
```
kotlin.AssertionError: Expected <97>, actual <-2145945247>.
```

Reply:
Interesting, I haven't seen this locally. It also passed on iOS, watchOS, and tvOS. Looks like a corrupted pointer coming from the `NSOutputStream`'s `NSStreamDataWrittenToMemoryStreamKey` `NSData` property. Now the question is why would that memory have been freed or was the data actually empty or was it corrupted sometime before? [I added](https://github.com/Kotlin/kotlinx-io/pull/174/files#diff-0dac6639d2e10ace0d178ca7cdfb08b54e570eba7dbf81cabf4603db62349cd4R22-R24) an additional check for the `NSData` length being 1 and changed the code accessing the byte from the byte array.

Reply:
I guess the issue caused by the reinterpret cast from int8* to int32*: the read following it fetches the expected single byte and three bytes after it.
Note that `-2145945247` is `0x80184961`, and given little-endian ordering on X86-64, the test reads:
```
0x61 0x49 0x18 0x80
  ^   ^
  |   |
  |   Garbage after it
  |
The value you're trying to verify
```

Reply:
Ah, yes! Here I am writing a byte and reading four of them. Fixed that.